nshlib/nsh_codeccmd.c

```diff
@@ -443,8 +443,8 @@ static int cmd_codecs_proc(FAR struct nsh_vtbl_s *vtbl, int argc,
           int i;
 
           md5_final(mac, &ctx);
-          src = (FAR char *)&mac;
-          dest = destbuf;
+          src  = mac;
+          dest = (FAR unsigned char *)destbuf;
```

Comment:
we can keep dest as `char`, only src need change to `unsigned char` to avoid the array index become negative in `hexchars[(*src) >> 4]`.